Defining Essential Models and their Relationships

YarekTenko opened the pull request:
Established associations between User, Bet and Option models through UserBets join table.

qr8r commented on db/migrate/20170202234757_add_credits_to_users.rb:
```diff
@@ -0,0 +1,5 @@
+class AddCreditsToUsers < ActiveRecord::Migration[5.0]
+  def change
+    add_column :users, :credits, :integer
```
For now lets assume that all users have the money to back up bets. The system doesnt need to track how many credits a user has

qr8r commented on db/migrate/20170127000826_add_outcomes.rb:
```diff
@@ -1,11 +0,0 @@
-class AddOutcomes < ActiveRecord::Migration[5.0]
```
Dont delete existing migrations. To change existing structure, add new migrations

qr8r commented on app/models/outcome.rb:
```diff
@@ -1,4 +0,0 @@
-class Outcome < ApplicationRecord
-  belongs_to :bet
```
Doesnt belong in this commit

qr8r commented on app/models/user.rb:
```diff
@@ -1,4 +1,6 @@
+  has_many :bets, through: :user_bets
+  has_many :user_bets
+
+  has_secure_password
```
From the session branch

qr8r commented on db/migrate/20170202232146_add_option_total_to_options.rb:
```diff
@@ -0,0 +1,5 @@
+class AddOptionTotalToOptions < ActiveRecord::Migration[5.0]
+  def change
```
These migrations add colums to the same model in the same commit. It can probably be the same migration

qr8r commented on app/models/option.rb:
```diff
@@ -0,0 +1,4 @@
+class Option < ApplicationRecord
```
Option is a very generic name for this. Now that we have changed what used to be `Outcomes` to `UserBet` I think outcome more accurately describes what this model represents

qr8r commented on db/schema.rb:
```diff
@@ -20,6 +20,13 @@
+    t.datetime "created_at",   null: false
+    t.datetime "updated_at",   null: false
+    t.text     "option_text"
+    t.integer  "option_total"
```
So every user who bets on the same option (outcome) has to bet the same amount of money ?

qr8r commented on app/models/user_bet.rb:
```diff
@@ -0,0 +1,5 @@
+class UserBet < ApplicationRecord
+  belongs_to :user
```
Should be in the same commit as when you remove the Outcomes model earlier

qr8r commented on db/migrate/20170202212027_create_user_bets.rb:
```diff
@@ -0,0 +1,8 @@
+class CreateUserBets < ActiveRecord::Migration[5.0]
+  def change
```
all of these migrations should be merged into a single migration

qr8r commented on db/migrate/20170202212555_add_user_to_user_bets.rb:
```diff
@@ -0,0 +1,5 @@
+class AddUserToUserBets < ActiveRecord::Migration[5.0]
+  def change
+    add_reference :user_bets, :user, foreign_key: true
```
dont forget to index references as well

qr8r reviewed (changes requested):
These commits are well sized but they are not coherent. Some commits contain code they shouldn't need.

Another rule of thumb is that the project should be 'functional' after every commit. So that I could checkout any commit you make and the project would still run without errors

LeonTenko committed:
Add BetOption model

-Created BetOption model;
-Defined many-to-many association for Bet with BetOption through
UserBet;
-Defined belongs_to association for UserBet with BetOption.

LeonTenko committed:
Add creator_id column to Bet model

LeonTenko committed:
Rename Outcome model to UserBet
